GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -524,7 +524,7 @@ public override void Refresh()
                 _lastVisibleResizableColumn.Resizable = DataGridViewTriState.True;
             }
 
-            _gridView.Refresh(); // columns could change their Resizable state, e.g. the BuildStatusColumnProvider
+            _gridView.ApplySettings(); // columns could change their Resizable state, e.g. the BuildStatusColumnProvider
```

Comment:
My only feedback is that it is not immediately apparent here what "settings" we're applying here. Semantically, "refresh" was more meaningful, i.e., its function more apparent to the reader (I'm not saying it was _right_ though).

The commit message is very light on details as to why the split was necessary. I have some guesses but those are rather speculative.
It would be great if you could provide the background that led you to the refactor as well as detail how `RevisionDataGridView.Refresh` is being invoked now.

Reply:
I have updated the PR description with the former design issues and with the benefits.

`RevisionGridControl` calls `RevisionDataGridView.Refresh` if the grid needs to be rendered anew because the list of revisions has changed.
`RevisionDataGridView.ApplySettings` is initially called once and if settings have changed. It finally calls `Refresh` - IMO "of course" because "Apply" includes the refresh.
Rename to `LoadSettingsAndRefresh`? But then rather separate calls `LoadSettings` and `Refresh`.
("Which settings?" is answered by that it is a method of `RevisionDataGridView`. Hence it means "all settings of this grid view".)

Reply:
Thank you for the explanation